language/move-prover/bytecode/src/mutation_tester.rs

```diff
@@ -78,28 +117,100 @@ impl FunctionTargetProcessor for MutationTester {
         for bc in code {
             match bc {
                 Call(ref attrid, ref indices, Operation::Add, ref srcs, ref dests) => {
+                    let call = Call(
+                        *attrid,
+                        (*indices).clone(),
+                        Operation::Sub,
+                        (*srcs).clone(),
+                        (*dests).clone(),
+                    );
                     let mv = m.add_sub;
-                    if mv == 1 {
-                        builder.emit(Call(
-                            *attrid,
-                            (*indices).clone(),
-                            Operation::Sub,
-                            (*srcs).clone(),
-                            (*dests).clone(),
-                        ));
-                        global_env.set_extension(MutationManager {
-                            mutated: true,
-                            add_sub: mv - 1,
-                        });
+                    let result: usize;
+                    if mv > 0 {
+                        result = mv - 1;
+                    } else {
+                        result = mv;
+                    }
+                    let mm = MutationManager {
```

Comment:
You can here and elsewhere use the struct update syntax: `let mm = MutationManager{ add_sub.result, ..m}`.

Reply:
Fixed, thanks for the suggestion!